searchcore/src/vespa/searchcore/proton/documentmetastore/documentmetastore.cpp

```diff
@@ -462,20 +462,23 @@ DocumentMetaStore::put(const GlobalId &gid,
                        const BucketId &bucketId,
                        const Timestamp &timestamp,
                        uint32_t docSize,
-                       DocId lid)
+                       DocId lid,
+                       uint64_t prepare_serial_num)
 {
     Result res;
     RawDocumentMetaData metaData(gid, bucketId, timestamp, docSize);
     KeyComp comp(metaData, _metaDataStore, *_gidCompare);
-    TreeType::Iterator itr = _gidToLidMap.lowerBound(KeyComp::FIND_DOC_ID,
-            comp);
+    auto& itr = _gid_to_lid_map_write_itr;
+    if (prepare_serial_num == 0u || _gid_to_lid_map_write_itr_prepare_serial_num != prepare_serial_num) {
```

Comment:
Consider adding a helper function for this, e.g. update_gid_to_lid_map_write_itr(prepare_serial_num). Same check is used in three places.